### DISCUSSION TITLE
Runtime generation of routes

### DISCUSSION THREAD

**Discussion by jafioti** [rwf2/Rocket#1919] [Questions] [answered]
Hi,

Is it possible to generate a new route at runtime and mount it? When my program is first ran, I would like to read blog posts from a database, generate routes for each of them, and mount them. Mounting is straightforward but I can't seem to generate the functions for the routes because it seems that procedural macros aren't allowed on functions in functions. 

For instance, this doesn't work: 
```
fn create_route(name: String, rocket: Rocket<Build>) -> Rocket<Build> {
    #[get(/url)]
    fn new_route() -> Markup {
        html!{
            p {"Hello " (name)}
        }
    }

    rocket.mount("/", routes![new_route()])
}
```

I previously simply had a single route for the blog posts with a dynamic url so that when they visited a page, it would show the correct one based on the url. These pages seem to not get caught by google though, so I figured I should make each of them actual routes, so they get picked up for SEO.

Thanks for the help!

**Comment by the10thWiz** [comment, marked as answer]
The mounting of routes shouldn't effect SEO at all, since google shouldn't be able to tell how you mounted them.

You likely have a different issue that's preventing Google from recommending your blog posts.

**Reply by jafioti** [reply]
I wasn't mounting the routes at all, rather dynamically serving them based on the dynamic path to the blog post. Since they were never mounted, google wasn't able to see them.

There was only one single blog route, with a dynamic path, and when the user visits that path, plus some blog post name, it will try to retrieve that blog post and display it.

**Reply by the10thWiz** [reply]
My point is that Google should be able to automatically pick up on these routes anyway. Google accesses routes exactly the same as a user. There may be other issues, i.e. Google doesn't think these pages are worth showing in results, or you don't have links to them anywhere else so Google can't find them.

**Reply by jafioti** [reply]
Ah, I see. So google just goes through your pages, looks for links to other pages, and repeats. Ok, I put in links to the dynamically generated pages, thanks for the help!
